scheduler/kubernetes/upgrade.md

```diff
@@ -8,57 +8,115 @@ sidebar: home_sidebar
+
+You can re-enter the parameters on the YAML web-form at [install.portworx.com](http://install.portworx.com), or convert them manually.
+The final YAML-spec from our example above would look similar to this:
+
+```bash
+VER=$(kubectl version --short | awk -Fv '/Server Version: /{print $3}')
+curl -o oci-spec.yaml \
+   'http://install.portworx.com?c=cluster123&k=etcd:http://etcd1.acme.net:2379,etcd:http://etcd2.acme.net:2379&s=/dev/sdb1,/dev/sdc&kbver=$VER'
+kubectl apply -f oci-spec.yaml
+```
+
+
+
+Once you have applied the new YAML-spec, Kubernetes will start applying the Portworx upgrade in a "RollingUpdate" fashion, one node at a time.
+
+
+### Step 2) Monitor the rolling upgrade
+
+<U>ROLLOUT STATUS</U>:<br/>
+One can monitor the upgrade process by running the "kubectl rollout status" command:
+
+```
+$ kubectl rollout status ds/portworx -n kube-system
+Waiting for rollout to finish: 0 out of 4 new pods have been updated...
+Waiting for rollout to finish: 1 out of 4 new pods have been updated...
+Waiting for rollout to finish: 2 out of 4 new pods have been updated...
+Waiting for rollout to finish: 3 out of 4 new pods have been updated...
+Waiting for rollout to finish: 3 of 4 updated pods are available...
+daemon set "portworx" successfully rolled out
+```
+
+Note that this command will inform us of general upgrade progress, but it will not point on which exact node is being upgraded and when.
+
+<U>NODES PORTWORX PODS STATUS</U>:<br/>
+To get more information about the status of Portworx daemonset across the nodes, we can run the following command:
+
+```
+$ kubectl get pods -o wide -n kube-system -l name=portworx
+NAME             READY     STATUS              RESTARTS   AGE       IP              NODE
+portworx-9njsl   1/1       Running             0          16d       192.168.56.73   minion4
+portworx-fxjgw   1/1       Running             0          16d       192.168.56.74   minion5
+portworx-fz2wf   1/1       Running             0          5m        192.168.56.72   minion3
+portworx-x29h9   0/1       ContainerCreating   0          0s        192.168.56.71   minion2
+```
+
+As we can see in the example output above:
+
+* looking at the STATUS/READY, we can tell that the rollout-upgrade is currently creating the container on the "minion2" node
+* looking at AGE, we can tell that
+	* "minion4" and "minion5" have Portworx up for 16 days (likely still on old version, and to be upgraded), while the
+	* "minion3" has Portworx up for only 5 minutes (likely just upgraded and restarted)
+* if we keep on monitoring, we will observe that the upgrade will not switch to the "next" node until STATUS is "Running" and the READY is 1/1 (meaning, the "readynessProbe" reports Portworx service is fully up).
+
+<U>PORTWORX CLUSTER LIST</U>:<br/>
+Finally, one can also run the following command to inspect the Portworx cluster:
+
+```
+minion1$ /opt/pwx/bin/pxctl cluster list
```

Comment:
Use `kubectl` method for running the pxctl cluster status command since many setups may not have ssh access to px nodes.